cppunit: enabling tests in libdnf leads to "Undefined symbols for architecture x86_64"

**Describe the bug**
As seen in https://github.com/NixOS/nixpkgs/pull/107348#discussion_r547528819 , trying to enable tests leads to a large amount of undefined symbols errors:

```
[100%] Linking CXX executable run_tests
Undefined symbols for architecture x86_64:
  "CppUnit::SourceLine::SourceLine(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int)", referenced from:
      ModuleProfileTest::testGetName()       in ModuleProfileTest.cpp.o
      ModuleProfileTest::testGetDescription()       in ModuleProfileTest.cpp.o
      ModuleProfileTest::testGetContent()       in ModuleProfileTest.cpp.o
      ModulePackageTest::testSimpleGetters()       in ModulePackageTest.cpp.o
      ModulePackageTest::testGetArtifacts()       in ModulePackageTest.cpp.o
      ModulePackageTest::testGetProfiles()       in ModulePackageTest.cpp.o
      ModulePackageTest::testGetModuleDependencies()       in ModulePackageTest.cpp.o
      ...
  "CppUnit::TextTestRunner::run(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, bool, bool)", referenced from:
      _main in run_tests.cpp.o
  "CppUnit::AdditionalMessage::AdditionalMessage(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)", referenced from:
      void CppUnit::assertEquals<int>(int const&, int const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in DependencyContainerTest.cpp.o
      void CppUnit::assertEquals<int>(int const&, int const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in CompsEnvironmentItemTest.cpp.o
      void CppUnit::assertEquals<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in CompsEnvironmentItemTest.cpp.o
      void CppUnit::assertEquals<int>(int const&, int const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in CompsGroupItemTest.cpp.o
      void CppUnit::assertEquals<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in CompsGroupItemTest.cpp.o
      void CppUnit::assertEquals<int>(int const&, int const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in TransactionItemReasonTest.cpp.o
      void CppUnit::assertEquals<libdnf::TransactionItemReason>(libdnf::TransactionItemReason const&, libdnf::TransactionItemReason const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in TransactionItemReasonTest.cpp.o
      ...
  "CppUnit::CompilerOutputter::CompilerOutputter(CppUnit::TestResultCollector*, std::basic_ostream<char, std::char_traits<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)", referenced from:
      _main in run_tests.cpp.o
  "CppUnit::TestFactoryRegistry::getRegistry(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)", referenced from:
      __GLOBAL__sub_I_ModuleProfileTest.cpp in ModuleProfileTest.cpp.o
      __GLOBAL__sub_I_ModulePackageTest.cpp in ModulePackageTest.cpp.o
      __GLOBAL__sub_I_ContextTest.cpp in ContextTest.cpp.o
      __GLOBAL__sub_I_ModulePackageContainerTest.cpp in ModulePackageContainerTest.cpp.o
      __GLOBAL__sub_I_PackageTest.cpp in PackageTest.cpp.o
      __GLOBAL__sub_I_DependencyTest.cpp in DependencyTest.cpp.o
      __GLOBAL__sub_I_DependencyContainerTest.cpp in DependencyContainerTest.cpp.o
      ...
  "CppUnit::Message::addDetail(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)", referenced from:
      TransactionTest::testInsert()       in TransactionTest.cpp.o
      TransactionTest::testInsertWithSpecifiedId()       in TransactionTest.cpp.o
      TransactionTest::testInsert() [clone .cold] in TransactionTest.cpp.o
      TransactionTest::testInsertWithSpecifiedId() [clone .cold] in TransactionTest.cpp.o
  "CppUnit::Message::Message(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)", referenced from:
      TransactionTest::testInsert()       in TransactionTest.cpp.o
      TransactionTest::testInsertWithSpecifiedId()       in TransactionTest.cpp.o
  "CppUnit::Message::Message(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)", referenced from:
      ModuleProfileTest::testGetName()       in ModuleProfileTest.cpp.o
      ModuleProfileTest::testGetDescription()       in ModuleProfileTest.cpp.o
      ModuleProfileTest::testGetContent()       in ModuleProfileTest.cpp.o
      ModulePackageTest::testSimpleGetters()       in ModulePackageTest.cpp.o
      ModulePackageTest::testGetArtifacts()       in ModulePackageTest.cpp.o
      ModulePackageTest::testGetProfiles()       in ModulePackageTest.cpp.o
      ModulePackageTest::testGetModuleDependencies()       in ModulePackageTest.cpp.o
      ...
  "CppUnit::Message::Message(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)", referenced from:
      MergedTransactionTest::testMerge()      in MergedTransactionTest.cpp.o
  "CppUnit::Asserter::failNotEqual(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, CppUnit::SourceLine const&, CppUnit::AdditionalMessage const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)", referenced from:
      void CppUnit::assertEquals<int>(int const&, int const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in DependencyContainerTest.cpp.o
      void CppUnit::assertEquals<int>(int const&, int const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in CompsEnvironmentItemTest.cpp.o
      void CppUnit::assertEquals<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in CompsEnvironmentItemTest.cpp.o
      void CppUnit::assertEquals<int>(int const&, int const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in CompsGroupItemTest.cpp.o
      void CppUnit::assertEquals<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in CompsGroupItemTest.cpp.o
      void CppUnit::assertEquals<int>(int const&, int const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in TransactionItemReasonTest.cpp.o
      void CppUnit::assertEquals<libdnf::TransactionItemReason>(libdnf::TransactionItemReason const&, libdnf::TransactionItemReason const&, CppUnit::SourceLine, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) in TransactionItemReasonTest.cpp.o
      ...
  "CppUnit::TestCase::TestCase(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)", referenced from:
      ModuleProfileTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in ModuleProfileTest.cpp.o
      ModulePackageTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in ModulePackageTest.cpp.o
      CppUnit::TestSuiteFactory<ContextTest>::makeTest()       in ContextTest.cpp.o
      ModulePackageContainerTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in ModulePackageContainerTest.cpp.o
      PackageTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in PackageTest.cpp.o
      DependencyTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in DependencyTest.cpp.o
      DependencyContainerTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in DependencyContainerTest.cpp.o
      ...
  "CppUnit::TestSuite::TestSuite(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)", referenced from:
      CppUnit::TestSuiteFactory<ModuleProfileTest>::makeTest()       in ModuleProfileTest.cpp.o
      CppUnit::TestSuiteFactory<ModulePackageTest>::makeTest()       in ModulePackageTest.cpp.o
      CppUnit::TestSuiteFactory<ContextTest>::makeTest()       in ContextTest.cpp.o
      CppUnit::TestSuiteFactory<ModulePackageContainerTest>::makeTest()       in ModulePackageContainerTest.cpp.o
      CppUnit::TestSuiteFactory<PackageTest>::makeTest()       in PackageTest.cpp.o
      CppUnit::TestSuiteFactory<DependencyTest>::makeTest()       in DependencyTest.cpp.o
      CppUnit::TestSuiteFactory<DependencyContainerTest>::makeTest()       in DependencyContainerTest.cpp.o
      ...
  "CppUnit::TestSuiteBuilderContextBase::getTestNameFor(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const", referenced from:
      ModuleProfileTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in ModuleProfileTest.cpp.o
      ModulePackageTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in ModulePackageTest.cpp.o
      CppUnit::TestSuiteFactory<ContextTest>::makeTest()       in ContextTest.cpp.o
      ModulePackageContainerTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in ModulePackageContainerTest.cpp.o
      PackageTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in PackageTest.cpp.o
      DependencyTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in DependencyTest.cpp.o
      DependencyContainerTest::addTestsToSuite(CppUnit::TestSuiteBuilderContextBase&)         in DependencyContainerTest.cpp.o
      ...
  "CppUnit::Test::findTestPath(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, CppUnit::TestPath&) const", referenced from:
      vtable for CppUnit::TestCaller<ModuleProfileTest> in ModuleProfileTest.cpp.o
      vtable for ModuleProfileTest in ModuleProfileTest.cpp.o
      vtable for CppUnit::TestCaller<ModulePackageTest> in ModulePackageTest.cpp.o
      vtable for ModulePackageTest in ModulePackageTest.cpp.o
      vtable for CppUnit::TestCaller<ContextTest> in ContextTest.cpp.o
      vtable for ContextTest in ContextTest.cpp.o
      vtable for CppUnit::TestCaller<ModulePackageContainerTest> in ModulePackageContainerTest.cpp.o
      ...
  "CppUnit::Test::resolveTestPath(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const", referenced from:
      vtable for CppUnit::TestCaller<ModuleProfileTest> in ModuleProfileTest.cpp.o
      vtable for ModuleProfileTest in ModuleProfileTest.cpp.o
      vtable for CppUnit::TestCaller<ModulePackageTest> in ModulePackageTest.cpp.o
      vtable for ModulePackageTest in ModulePackageTest.cpp.o
      vtable for CppUnit::TestCaller<ContextTest> in ContextTest.cpp.o
      vtable for ContextTest in ContextTest.cpp.o
      vtable for CppUnit::TestCaller<ModulePackageContainerTest> in ModulePackageContainerTest.cpp.o
      ...
  "CppUnit::Test::findTest(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const", referenced from:
      vtable for CppUnit::TestCaller<ModuleProfileTest> in ModuleProfileTest.cpp.o
      vtable for ModuleProfileTest in ModuleProfileTest.cpp.o
      vtable for CppUnit::TestCaller<ModulePackageTest> in ModulePackageTest.cpp.o
      vtable for ModulePackageTest in ModulePackageTest.cpp.o
      vtable for CppUnit::TestCaller<ContextTest> in ContextTest.cpp.o
      vtable for ContextTest in ContextTest.cpp.o
      vtable for CppUnit::TestCaller<ModulePackageContainerTest> in ModulePackageContainerTest.cpp.o
      ...
  "CppUnit::TestCase::getName[abi:cxx11]() const", referenced from:
      vtable for CppUnit::TestCaller<ModuleProfileTest> in ModuleProfileTest.cpp.o
      vtable for ModuleProfileTest in ModuleProfileTest.cpp.o
      vtable for CppUnit::TestCaller<ModulePackageTest> in ModulePackageTest.cpp.o
      vtable for ModulePackageTest in ModulePackageTest.cpp.o
      vtable for CppUnit::TestCaller<ContextTest> in ContextTest.cpp.o
      vtable for ContextTest in ContextTest.cpp.o
      vtable for CppUnit::TestCaller<ModulePackageContainerTest> in ModulePackageContainerTest.cpp.o
      ...
  "CppUnit::TestNamer::getFixtureName[abi:cxx11]() const", referenced from:
      CppUnit::TestSuiteFactory<ModuleProfileTest>::makeTest()       in ModuleProfileTest.cpp.o
      CppUnit::TestSuiteFactory<ModulePackageTest>::makeTest()       in ModulePackageTest.cpp.o
      CppUnit::TestSuiteFactory<ContextTest>::makeTest()       in ContextTest.cpp.o
      CppUnit::TestSuiteFactory<ModulePackageContainerTest>::makeTest()       in ModulePackageContainerTest.cpp.o
      CppUnit::TestSuiteFactory<PackageTest>::makeTest()       in PackageTest.cpp.o
      CppUnit::TestSuiteFactory<DependencyTest>::makeTest()       in DependencyTest.cpp.o
      CppUnit::TestSuiteFactory<DependencyContainerTest>::makeTest()       in DependencyContainerTest.cpp.o
      ...
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
make[2]: *** [tests/CMakeFiles/run_tests.dir/build.make:368: tests/run_tests] Error 1
make[1]: *** [CMakeFiles/Makefile2:833: tests/CMakeFiles/run_tests.dir/all] Error 2
make: *** [Makefile:160: all] Error 2
```

**To Reproduce**
Steps to reproduce the behavior:
1. Remove the postpatch in pkgs/tools/package-management/libdnf/default.nix
2. Build
3. Watch the missing symobl errors

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
libdnf
```

--- a/pkgs/development/python-modules/dnf4/default.nix
+++ b/pkgs/development/python-modules/dnf4/default.nix
@@ -1,0 +1,83 @@
+{ lib
+, buildPythonPackage
+, cmake
+, fetchFromGitHub
+, gettext
+, libcomps
+, libdnf
+, python
+, rpm
+, sphinx
+}:
+
+buildPythonPackage rec {
+  pname = "dnf4";
+  version = "4.18.1";
+  format = "other";
+
+  outputs = [ "out" "man" ];
+
+  src = fetchFromGitHub {
+    owner = "rpm-software-management";
+    repo = "dnf";
+    rev = version;
+    hash = "sha256-pgS4y87HbFiaS+fftdhKHOZAl2hYTUds3XVXUuQN6tU=";
+  };
+
+  patches = [
+    ./fix-python-install-dir.patch
+  ];
+
+  postPatch = ''
+    substituteInPlace CMakeLists.txt \
+      --replace "@PYTHON_INSTALL_DIR@" "$out/${python.sitePackages}" \
+      --replace "SYSCONFDIR /etc" "SYSCONFDIR $out/etc" \
+      --replace "SYSTEMD_DIR /usr/lib/systemd/system" "SYSTEMD_DIR $out/lib/systemd/system"
+    substituteInPlace etc/tmpfiles.d/CMakeLists.txt \
+      --replace "DESTINATION /usr/lib/tmpfiles.d" "DESTINATION $out/usr/lib/tmpfiles.d"
+    substituteInPlace dnf/const.py.in \
+      --replace "/etc" "$out/etc"
+    substituteInPlace doc/CMakeLists.txt \
+      --replace 'SPHINX_BUILD_NAME "sphinx-build-3"' 'SPHINX_BUILD_NAME "${sphinx}/bin/sphinx-build"'
+  '';
+
+  nativeBuildInputs = [
+    cmake
+    gettext
+    sphinx
+  ];
+
+  propagatedBuildInputs = [
+    libcomps
+    libdnf
+    rpm
+  ];
+
+  cmakeFlags = [
+    "-DPYTHON_DESIRED=${lib.head (lib.splitString ["."] python.version)}"
+  ];
+
+  postBuild = ''
+    make doc-man
+  '';
+
+  postInstall = ''
+    # See https://github.com/rpm-software-management/dnf/blob/41a287e2bd60b4d1100c329a274776ff32ba8740/dnf.spec#L218-L220
+    ln -s dnf-3 $out/bin/dnf
+    ln -s dnf-3 $out/bin/dnf4
+    mv $out/bin/dnf-automatic-3 $out/bin/dnf-automatic
+    # See https://github.com/rpm-software-management/dnf/blob/41a287e2bd60b4d1100c329a274776ff32ba8740/dnf.spec#L231-L232
+    ln -s $out/etc/dnf/dnf.conf $out/etc/yum.conf
+    ln -s dnf-3 $out/bin/yum
+  '';
+
+  meta = with lib; {
+    description = "Package manager based on libdnf and libsolv. Replaces YUM";
+    homepage = "https://github.com/rpm-software-management/dnf";
+    changelog = "https://github.com/rpm-software-management/dnf/releases/tag/${version}";
+    license = licenses.gpl2Only;
+    maintainers = with maintainers; [ katexochen ];
+    mainProgram = "dnf";
+    platforms = platforms.unix;
+  };
+}

--- a/pkgs/development/python-modules/dnf4/fix-python-install-dir.patch
+++ b/pkgs/development/python-modules/dnf4/fix-python-install-dir.patch
@@ -1,0 +1,13 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 4aee99fb..0bb28897 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -19,7 +19,7 @@ ELSE ()
+     MESSAGE (FATAL_ERROR "Invalid PYTHON_DESIRED value: " ${PYTHON_DESIRED})
+ ENDIF()
+
+-EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} -c "from sys import stdout; from sysconfig import get_path; stdout.write(get_path('purelib'))" OUTPUT_VARIABLE PYTHON_INSTALL_DIR)
++SET(PYTHON_INSTALL_DIR "@PYTHON_INSTALL_DIR@")
+ MESSAGE(STATUS "Python install dir is ${PYTHON_INSTALL_DIR}")
+
+ ADD_SUBDIRECTORY (dnf)

--- a/pkgs/tools/package-management/libcomps/default.nix
+++ b/pkgs/tools/package-management/libcomps/default.nix
@@ -1,0 +1,65 @@
+{ lib
+, check
+, cmake
+, doxygen
+, expat
+, fetchFromGitHub
+, libxml2
+, python
+, sphinx
+, stdenv
+, zlib
+}:
+
+stdenv.mkDerivation rec {
+  pname = "libcomps";
+  version = "0.1.20";
+
+  outputs = [ "out" "dev" "py" ];
+
+  src = fetchFromGitHub {
+    owner = "rpm-software-management";
+    repo = "libcomps";
+    rev = version;
+    hash = "sha256-IX4du1+G7lwWrGnllydnBDap2aqK5pzos1Mdyu4MzOU=";
+  };
+
+  patches = [
+    ./fix-python-install-dir.patch
+  ];
+
+  postPatch = ''
+    substituteInPlace libcomps/src/python/src/CMakeLists.txt \
+      --replace "@PYTHON_INSTALL_DIR@" "$out/${python.sitePackages}"
+  '';
+
+  nativeBuildInputs = [
+    check
+    cmake
+    doxygen
+    python
+    sphinx
+  ];
+
+  buildInputs = [
+    expat
+    libxml2
+    zlib
+  ];
+
+  dontUseCmakeBuildDir = true;
+  cmakeDir = "libcomps";
+
+  postFixup = ''
+    ls $out/lib
+    moveToOutput "lib/${python.libPrefix}" "$py"
+  '';
+
+  meta = with lib; {
+    description = "Comps XML file manipulation library";
+    homepage = "https://github.com/rpm-software-management/libcomps";
+    license = licenses.gpl2Only;
+    maintainers = with maintainers; [ katexochen ];
+    platforms = platforms.unix;
+  };
+}

--- a/pkgs/tools/package-management/libcomps/fix-python-install-dir.patch
+++ b/pkgs/tools/package-management/libcomps/fix-python-install-dir.patch
@@ -1,0 +1,13 @@
+diff --git a/libcomps/src/python/src/CMakeLists.txt b/libcomps/src/python/src/CMakeLists.txt
+index d22b84e..57bd1c2 100644
+--- a/libcomps/src/python/src/CMakeLists.txt
++++ b/libcomps/src/python/src/CMakeLists.txt
+@@ -85,7 +85,7 @@ IF (SKBUILD)
+     INSTALL(FILES libcomps/__init__.py DESTINATION libcomps/src/python/src/libcomps)
+     INSTALL(TARGETS pycomps LIBRARY DESTINATION libcomps/src/python/src/libcomps)
+ ELSE ()
+-    EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} -c "from sys import stdout; from sysconfig import get_path; stdout.write(get_path('platlib'))" OUTPUT_VARIABLE PYTHON_INSTALL_DIR)
++    SET(PYTHON_INSTALL_DIR "@PYTHON_INSTALL_DIR@")
+
+     INSTALL(FILES ${pycomps_SRCDIR}/libcomps/__init__.py DESTINATION ${PYTHON_INSTALL_DIR}/libcomps)
+     #INSTALL(FILES ${pycomps_SRCDIR}/tests/__test.py DESTINATION

--- a/pkgs/tools/package-management/libdnf/default.nix
+++ b/pkgs/tools/package-management/libdnf/default.nix
@@ -99,6 +99,6 @@ stdenv.mkDerivation rec {
     changelog = "https://github.com/rpm-software-management/libdnf/releases/tag/${version}";
     license = licenses.gpl2Plus;
     platforms = platforms.linux ++ platforms.darwin;
-    maintainers = with maintainers; [ rb2k ];
+    maintainers = with maintainers; [ rb2k katexochen ];
   };
 }

--- a/pkgs/tools/package-management/libdnf/default.nix
+++ b/pkgs/tools/package-management/libdnf/default.nix
@@ -16,11 +16,17 @@
 , rpm
 , zchunk
 , cppunit
+, python
+, swig
+, glib
+, sphinx
 }:
 
 stdenv.mkDerivation rec {
   pname = "libdnf";
   version = "0.72.0";
+
+  outputs = [ "out" "dev" "py" ];
 
   src = fetchFromGitHub {
     owner = "rpm-software-management";
@@ -44,6 +50,9 @@ stdenv.mkDerivation rec {
     libyaml
     libmodulemd
     zchunk
+    python
+    swig
+    sphinx
   ];
 
   propagatedBuildInputs = [
@@ -58,17 +67,31 @@ stdenv.mkDerivation rec {
     cp ${libsolv}/share/cmake/Modules/FindLibSolv.cmake cmake/modules/
   '';
 
+  patches = [
+    ./fix-python-install-dir.patch
+  ];
+
   postPatch = ''
     # https://github.com/rpm-software-management/libdnf/issues/1518
     substituteInPlace libdnf/libdnf.pc.in \
       --replace '$'{prefix}/@CMAKE_INSTALL_LIBDIR@ @CMAKE_INSTALL_FULL_LIBDIR@
+    substituteInPlace cmake/modules/FindPythonInstDir.cmake \
+      --replace "@PYTHON_INSTALL_DIR@" "$out/${python.sitePackages}"
   '';
 
   cmakeFlags = [
     "-DWITH_GTKDOC=OFF"
     "-DWITH_HTML=OFF"
-    "-DWITH_BINDINGS=OFF"
+    "-DPYTHON_DESIRED=${lib.head (lib.splitString ["."] python.version)}"
   ];
+
+  postInstall = ''
+    rm -r $out/${python.sitePackages}/hawkey/test
+  '';
+
+  postFixup = ''
+    moveToOutput "lib/${python.libPrefix}" "$py"
+  '';
 
   meta = with lib; {
     description = "Package management library";

--- a/pkgs/tools/package-management/libdnf/default.nix
+++ b/pkgs/tools/package-management/libdnf/default.nix
@@ -15,6 +15,7 @@
 , libyaml
 , rpm
 , zchunk
+, cppunit
 }:
 
 stdenv.mkDerivation rec {
@@ -36,6 +37,7 @@ stdenv.mkDerivation rec {
 
   buildInputs = [
     check
+    cppunit
     openssl
     json_c
     libsmartcols
@@ -57,11 +59,6 @@ stdenv.mkDerivation rec {
   '';
 
   postPatch = ''
-    # See https://github.com/NixOS/nixpkgs/issues/107428
-    substituteInPlace CMakeLists.txt \
-      --replace "enable_testing()" "" \
-      --replace "add_subdirectory(tests)" ""
-
     # https://github.com/rpm-software-management/libdnf/issues/1518
     substituteInPlace libdnf/libdnf.pc.in \
       --replace '$'{prefix}/@CMAKE_INSTALL_LIBDIR@ @CMAKE_INSTALL_FULL_LIBDIR@

--- a/pkgs/tools/package-management/libdnf/fix-python-install-dir.patch
+++ b/pkgs/tools/package-management/libdnf/fix-python-install-dir.patch
@@ -1,0 +1,12 @@
+diff --git a/cmake/modules/FindPythonInstDir.cmake b/cmake/modules/FindPythonInstDir.cmake
+index ed098ded..2a2e1543 100644
+--- a/cmake/modules/FindPythonInstDir.cmake
++++ b/cmake/modules/FindPythonInstDir.cmake
+@@ -1,6 +1 @@
+-EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} -c "
+-from sys import stdout
+-from sysconfig import get_path
+-path=get_path(name='platlib', vars={'platbase':'${CMAKE_INSTALL_PREFIX}'})
+-stdout.write(path)"
+-OUTPUT_VARIABLE PYTHON_INSTALL_DIR)
++SET(PYTHON_INSTALL_DIR "@PYTHON_INSTALL_DIR@")

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -605,6 +605,8 @@ with pkgs;
 
   dec-decode = callPackage ../development/tools/dec-decode { };
 
+  dnf4 =  with python3Packages; toPythonApplication dnf4;
+
   dnf5 = callPackage ../tools/package-management/dnf5 { };
 
   documenso = callPackage ../applications/office/documenso { };

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -22570,6 +22570,8 @@ with pkgs;
 
   libcollectdclient = callPackage ../development/libraries/libcollectdclient { };
 
+  libcomps = callPackage ../tools/package-management/libcomps { python = python3; };
+
   libcpr = callPackage ../development/libraries/libcpr { };
 
   libcredis = callPackage ../development/libraries/libcredis { };

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -22658,7 +22658,7 @@ with pkgs;
 
   libdnet = callPackage ../development/libraries/libdnet { };
 
-  libdnf = callPackage ../tools/package-management/libdnf { };
+  libdnf = callPackage ../tools/package-management/libdnf { python = python3; };
 
   libdovi = callPackage ../development/libraries/libdovi { };
 

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -3204,6 +3204,8 @@ self: super: with self; {
 
   dnachisel = callPackage ../development/python-modules/dnachisel { };
 
+  dnf4 = callPackage ../development/python-modules/dnf4 { };
+
   dnfile = callPackage ../development/python-modules/dnfile { };
 
   dnslib = callPackage ../development/python-modules/dnslib { };

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -6139,6 +6139,13 @@ self: super: with self; {
 
   libcst = callPackage ../development/python-modules/libcst { };
 
+  libdnf = lib.pipe pkgs.libdnf [
+    toPythonModule
+    (p: p.overrideAttrs (super: { meta = super.meta // { outputsToInstall = [ "py" ]; }; }))
+    (p: p.override { inherit python; })
+    (p: p.py)
+  ];
+
   libevdev = callPackage ../development/python-modules/libevdev { };
 
   libfdt = toPythonModule (pkgs.dtc.override {

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -6137,6 +6137,13 @@ self: super: with self; {
 
   libcloud = callPackage ../development/python-modules/libcloud { };
 
+  libcomps = lib.pipe pkgs.libcomps [
+    toPythonModule
+    (p: p.overrideAttrs (super: { meta = super.meta // { outputsToInstall = [ "py" ]; }; }))
+    (p: p.override { inherit python; })
+    (p: p.py)
+  ];
+
   libcst = callPackage ../development/python-modules/libcst { };
 
   libdnf = lib.pipe pkgs.libdnf [

